Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/Components/MonthGridController.cs

```diff
@@ -0,0 +1,87 @@
+using DCL.InWorldCamera.CameraReelStorageService.Schemas;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using UnityEngine;
+
+namespace DCL.InWorldCamera.CameraReelGallery.Components
+{
+    public class MonthGridController : IDisposable
+    {
+        internal readonly MonthGridView view;
+
+        private readonly List<ReelThumbnailController> reelThumbnailViews = new ();
+        private readonly ReelGalleryPoolManager reelGalleryPoolManager;
+
+        public DateTime DateTimeBucket { get; private set; }
+
+        public MonthGridController(MonthGridView view,
+            ReelGalleryPoolManager reelGalleryPool)
+        {
+            this.view = view;
+            this.reelGalleryPoolManager = reelGalleryPool;
+        }
+
+        public List<ReelThumbnailController> Setup(DateTime bucket,
+            List<CameraReelResponseCompact> images,
+            OptionButtonController optionsButton,
+            Action<CameraReelResponseCompact, Sprite> onThumbnailLoaded,
+            Action<CameraReelResponseCompact> onThumbnailClicked)
+        {
+            this.DateTimeBucket = bucket;
+            view.monthText.SetText(bucket.ToString("MMMM yyyy", CultureInfo.InvariantCulture));
+
+            List<ReelThumbnailController> newViews = new();
+
+            for (int i = 0; i < images.Count; i++)
+            {
+                ReelThumbnailController thumbnailView = reelGalleryPoolManager.GetThumbnailElement(images[i], view.gridLayoutGroup, optionsButton);
+                thumbnailView.ThumbnailLoaded += onThumbnailLoaded;
+                thumbnailView.ThumbnailClicked += onThumbnailClicked;
+                newViews.Add(thumbnailView);
+            }
+
+            reelThumbnailViews.AddRange(newViews);
+
+            return newViews;
+        }
+
+        public void OnPoolGet() =>
```

Comment:
Please remove the `On` prefix on methods. We only include them if they are directly linked to an event